OpenSSL_version.c

```diff
@@ -1,38 +1,55 @@
+/*!*****************************************************************************
+ * @file   OpenSSL_version.c
+ * @brief  OpenSSL version checking utility
+ * keep this file identical among genCMPClient and CMPforOpenSSL
+ *
+ * @author David von Oheimb, David.von.Oheimb@siemens.com
+ *
+ *  Copyright (c) 2021-2024 Siemens AG
+ *  Licensed under the Apache License, Version 2.0
+ *  SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <openssl/crypto.h>
 
-#if OPENSSL_VERSION_NUMBER < 0x10002000L
-# error Better not use OpenSSL versions older than 1.0.2. They are unsupported and insecure.
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
+    #error Should not use OpenSSL versions older than 3. They are unsupported and insecure.
```

Comment:
I would suggest to use "less than 3.0" instead of "older than 3".

Reply:
Well, the phrase "version(s) older than ..." is not uncommon,
but to avoid any misinterpretation, changed to "before" (which is also shorter).